src/runner/patch-round.js

```diff
@@ -0,0 +1,110 @@
+import { BIG_DECIMAL, DECIMAL_128 } from "../constants.js";
+
+// NOTE: I think we should still stick to the modes named in the
+// proposal and these could be cleaned up a little, but this is for
+// demonstration for now
+const internalModes = [null, "trunc", "ceil", "floor", "halfExpand"];
+const listFormatter = new Intl.ListFormat("en", {
+  style: "long",
+  type: "disjunction",
+});
+const modesList = listFormatter.format(internalModes.filter(Boolean));
+
+const roundingModeImpl = {
+  [DECIMAL_128](mode) {
+    const retval = internalModes.indexOf(`${mode}`);
+    if (mode === null || retval === -1)
+      throw new RangeError(`bad rounding mode "${mode}"`);
+    return retval;
+  },
+  [BIG_DECIMAL](mode, isNegative) {
+    switch (mode) {
+      case "ceil":
+        return isNegative ? Big.roundDown : Big.roundUp;
+      case "floor":
+        return isNegative ? Big.roundUp : Big.roundDown;
```

Comment:
It's true that we are now explicitly calling `Big` when we are in `Big` mode. I think making this clear is good for folks who want to look at the generated code and know what's happening. It also means that when we delete all of one case, once we agree on an approach, these will all go away, and if they don't, it's a good sign something was missed.